crates/re_viewer/src/ui/view_spatial/scene/scene_part/images.rs

```diff
@@ -146,64 +151,171 @@ impl ImagesPart {
-                    tensor,
-                    meter,
-                    annotations,
-                });
-            }
+        let color = annotations.class_description(None).annotation_info().color(
+            color.map(|c| c.to_array()).as_ref(),
+            DefaultColor::OpaqueWhite,
+        );
+
+        let highlight = entity_highlight.index_highlight(instance_path_hash.instance_key);
+        if highlight.is_some() {
+            let color = SceneSpatial::apply_hover_and_selection_effect_color(
+                re_renderer::Color32::TRANSPARENT,
+                highlight,
+            );
+            let rect = glam::vec2(tensor.shape()[1].size as f32, tensor.shape()[0].size as f32);
+            scene
+                .primitives
+                .line_strips
+                .batch("image outlines")
+                .world_from_obj(world_from_obj)
+                .add_axis_aligned_rectangle_outline_2d(glam::Vec2::ZERO, rect)
+                .color(color)
+                .radius(Size::new_points(1.0));
         }
 
-        Ok(())
+        push_tensor_texture(
+            scene,
+            ctx,
+            &annotations,
+            world_from_obj,
+            instance_path_hash,
+            &tensor,
+            color.into(),
+        );
+
+        // TODO(jleibs): Meter should really be its own component
+        let meter = tensor.meter;
+
+        scene.ui.images.push(Image {
+            instance_path_hash,
+            tensor,
+            meter,
+            annotations,
+        });
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn process_entity_view_as_depth_cloud(
+        scene: &mut SceneSpatial,
+        ctx: &mut ViewerContext<'_>,
+        transforms: &TransformCache,
+        properties: &EntityProperties,
+        tensor: &Tensor,
+        pinhole_ent_path: &EntityPath,
+    ) {
+        crate::profile_function!();
+
+        let Some(re_log_types::Transform::Pinhole(intrinsics)) = query_latest_single::<Transform>(
+            &ctx.log_db.entity_db,
+            pinhole_ent_path,
+            &ctx.current_query(),
+        ) else {
+            re_log::warn_once!("Couldn't fetch pinhole intrinsics at {pinhole_ent_path:?}");
+            return;
+        };
+
+        // TODO(cmc): getting to those extrinsics is no easy task :|
+        let Some(extrinsics) = pinhole_ent_path
+            .parent()
+            .and_then(|ent_path| transforms.reference_from_entity(&ent_path)) else {
+            re_log::warn_once!("Couldn't fetch pinhole extrinsics at {pinhole_ent_path:?}");
+            return;
+        };
+
+        // TODO(cmc): automagically convert as needed for non-natively supported datatypes?
+        let data = match &tensor.data {
+            // TODO(cmc): re_renderer needs native interop with Arrow, we shouldn't need to
+            // allocate this massive buffer and do a memcpy when we could just carry arround the
+            // original `arrow::Buffer`.
```

Comment:
but didn't you do this already on another PR that landed?

Reply:
not quite landed yet, it requires this PR!